src/auth_service/claims_repository/rest/router.py

```diff
@@ -447,18 +459,31 @@ async def get_datasets_with_download_access(
     if not await user_is_active(user_id, user_dao=user_dao):
         raise user_not_found_error
 
-    # fetch all valid controlled access grants for the user
-    dataset_ids = [
-        dataset_id_for_download_access(claim)
-        async for claim in claim_dao.find_all(
-            mapping={
-                "user_id": user_id,
-                "visa_type": VisaType.CONTROLLED_ACCESS_GRANTS,
-            }
-        )
-        if is_valid_claim(claim)
-        and claim.iva_id
-        and await iva_is_verified(user_id, claim.iva_id, iva_dao=iva_dao)
-    ]
-    # filter out datasets from different sources and sort for reproducible output
-    return sorted(dataset_id for dataset_id in dataset_ids if dataset_id)
+    dataset_ids: dict[str, UTCDatetime] = {}
```

Comment:
```suggestion
    dataset_expiries: dict[str, UTCDatetime] = {}
```
The type-hint makes it clear, but a clearer variable name would help maintain the clarity below, I think.

Reply:
Fixed in 4281aa1.